drivers/ethernet/eth_mcux.c

```diff
@@ -663,51 +663,6 @@ static bool eth_get_ptp_data(struct net_if *iface, struct net_pkt *pkt,
 
 	net_pkt_set_priority(pkt, NET_PRIORITY_CA);
 
-	if (ptpTsData) {
-		/* Cannot use GPTP_HDR as net_pkt fields are not all filled */
-		struct gptp_hdr *hdr;
-
-		/* In TX, the first net_buf contains the Ethernet header
-		 * and the actual gPTP header is in the second net_buf.
-		 * In RX, the Ethernet header + other headers are in the
-		 * first net_buf.
-		 */
-		if (is_tx) {
-			if (pkt->frags->frags == NULL) {
-				return false;
-			}
-
-			hdr = (struct gptp_hdr *)pkt->frags->frags->data;
-		} else {
-			hdr = (struct gptp_hdr *)(pkt->frags->data +
-							   eth_hlen);
-		}
-
-		ptpTsData->version = hdr->ptp_version;
-		memcpy(ptpTsData->sourcePortId, &hdr->port_id,
-		       kENET_PtpSrcPortIdLen);
-		ptpTsData->messageType = hdr->message_type;
-		ptpTsData->sequenceId = ntohs(hdr->sequence_id);
-
-#ifdef CONFIG_ETH_MCUX_PHY_EXTRA_DEBUG
-		LOG_DBG("PTP packet: ver %d type %d len %d seq %d",
-			ptpTsData->version,
-			ptpTsData->messageType,
-			ntohs(hdr->message_length),
-			ptpTsData->sequenceId);
-		LOG_DBG("  clk %02x%02x%02x%02x%02x%02x%02x%02x port %d",
-			hdr->port_id.clk_id[0],
-			hdr->port_id.clk_id[1],
-			hdr->port_id.clk_id[2],
-			hdr->port_id.clk_id[3],
-			hdr->port_id.clk_id[4],
-			hdr->port_id.clk_id[5],
-			hdr->port_id.clk_id[6],
-			hdr->port_id.clk_id[7],
-			ntohs(hdr->port_id.port_number));
-#endif
-	}
-
```

Comment:
You are removing lot of code here. Is gPTP and timestamping support really tested properly after this change?
I am asking because `frdm-k64f` has been one of the reference devices that support time sync in TSN and it would be important that gPTP continues to work properly in the future with this hw.

Reply:
I am not clear how to run the gPTP tests. Can you give me some guidance. I tried following the instructions provided, however I see an error `Group ptp not found` when running `sudo ./gptp` command

Reply:
One option for testing is to have Linux <-> Zephyr gPTP connectivity and for that I suggest linuxptp from http://linuxptp.sourceforge.net/. Linuxptp supports software timestamping so it can be run in normal PC.

The error `Group ptp not found` sounds like the ptp group is missing and your account is not part of that group.

The other option is to have two Zephyr devices that support gPTP and connect them together. Direct connection might work too, but if not then you need to connect them both to a Ethernet switch.

Reply:
Thanks, I have installed `linuxptp`. Can you provide some guidance on how to use it for testing frdm-k64F. I have the `samples/net/gptp` application running on the frdm-k64F board

Reply:
I tried with 2 boards running `samples/net/gptp`. I see the below printed on both boards and nothing more.
```
[00:00:00.269,000] <wrn> net_if: iface 0x80000f60 is down
[00:00:00.273,000] <inf> net_config: Initializing network
[00:00:00.273,000] <inf> net_config: Waiting interface 1 (0x80000f60) to be up...
[00:00:01.271,000] <wrn> net_gptp: Reset Pdelay requests
[00:00:01.273,000] <wrn> net_if: iface 0x80000f60 is down
uart:~$
```

Reply:
You can use `net gptp` and `net gptp 1` net-shell commands to see some information about gPTP status.

Reply:
I  am unable to type anything in the serial terminal window

Reply:
in my testing there are some problem
![image](https://user-images.githubusercontent.com/800198/96839731-28ec3100-147c-11eb-8784-35ca8de13988.png)

I run the test script in windows
.gptp <mac address> -F gptp_cfg.ini

@jukkar  can you have a check on this? looks like the gptp application is not work well in ptpv2 IEEE1588

Reply:
> @jukkar can you have a check on this? looks like the gptp application is not work well in ptpv2 IEEE1588

My frdm-k64f board just refuses the take the Ethernet link up so I am seeing the same issue as @mmahadevan108 few comments above.

Instead I used Atmel sam-e70-xplained board against Avnu gPTP daemon running in Linux, and everything is working fine there. Zephyr becomes slave and linux host became master, time sync is ok.
Can you do your tests without the lines 666-710 removed (I commented these above)? The removal of those lines looks suspicious and I would like to get confirmation whether they break gPTP or not.

Reply:
comment out those lines does not help, I still get the same result.  and the board console output is

```
[00:06:55.751,000] <wrn> net_gptp: Reset Pdelay requests
[00:06:56.756,000] <wrn> net_gptp: Reset Pdelay requests
[00:06:57.760,000] <wrn> net_gptp: Reset Pdelay requests
[00:06:58.764,000] <wrn> net_gptp: Reset Pdelay requests
[00:06:59.769,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:00.773,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:01.777,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:02.782,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:03.786,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:04.790,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:05.794,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:06.799,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:07.803,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:08.807,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:09.812,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:10.816,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:11.820,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:12.825,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:13.829,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:14.833,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:15.837,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:16.842,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:17.846,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:18.850,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:19.855,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:20.859,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:21.863,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:22.868,000] <wrn> net_gptp: Reset Pdelay requests
[00:07:23.872,000] <wrn> net_gptp: Reset Pdelay requests

```

Reply:
board networking config

```
[00:00:00.003,000] <wrn> net_if: iface 0x20000f40 is down
[00:00:00.007,000] <inf> net_config: Initializing network
[00:00:00.007,000] <inf> net_config: Waiting interface 1 (0x20000f40) to be up...
[00:00:01.005,000] <wrn> net_gptp: Reset Pdelay requests
[00:00:01.008,000] <wrn> net_if: iface 0x20000f40 is down
[00:00:02.010,000] <wrn> net_gptp: Reset Pdelay requests
[00:00:02.012,000] <wrn> net_if: iface 0x20000f40 is down
[00:00:03.001,000] <inf> net_config: Interface 1 (0x20000f40) coming up
[00:00:03.001,000] <inf> eth_mcux: ETH_0 enabled 100M full-duplex mode.
[00:00:03.001,000] <inf> net_config: IPv4 address: 192.168.0.103
[00:00:03.014,000] <wrn> net_gptp: Reset Pdelay requests
[00:00:03.102,000] <inf> net_config: IPv6 address: 2001:db8::1
uart:~$ net gptp 1
Port id    : 1 (DISABLED)
Interface  : 0x20000f40 [1]
Clock id   : 02:04:9f:ff:fe:31:96:09
Version    : 2
AS capable : no

Configuration:
Time synchronization and Best Master Selection enabled        : yes
The port is measuring the path delay                          : no
One way propagation time on the link attached to this port    : 0 ns
Propagation time threshold for the link attached to this port : 100000 ns
Estimate of the ratio of the frequency with the peer          : 1
Asymmetry on the link relative to the grand master time base  : %ld
Maximum interval between sync messages                        : %lu
Maximum number of Path Delay Requests without a response      : 3
Current Sync sequence id for this port                        : 24570
Current Path Delay Request sequence id for this port          : 50891
Current Announce sequence id for this port                    : 15514
Current Signaling sequence id for this port                   : 22146
Whether neighborRateRatio needs to be computed for this port  : yes
Whether neighborPropDelay needs to be computed for this port  : yes
Initial Announce Interval as a Logarithm to base 2            : 0
Current Announce Interval as a Logarithm to base 2            : 0
Initial Sync Interval as a Logarithm to base 2                : -4
Current Sync Interval as a Logarithm to base 2                : -4
Initial Path Delay Request Interval as a Logarithm to base 2  : 0
Current Path Delay Request Interval as a Logarithm to base 2  : 0
Time without receiving announce messages before running BMCA  : 1 ms (3)
Time without receiving sync messages before running BMCA      : %lu ms (1)
Sync event transmission interval for the port                 : %lu ms
Path Delay Request transmission interval for the port         : %lu ms
BMCA default priority1                                        : 248
BMCA default priority2                                        : 248

Runtime status:
Current global port state                                : DISABLED
Path Delay Request state machine variables:
Current state                                    : WAIT_RESP
Initial Path Delay Response Peer Timestamp       : %lu
Initial Path Delay Response Ingress Timestamp    : %lu
Path Delay Response messages received            : 0
Path Delay Follow Up messages received           : 0
Number of lost Path Delay Responses              : 3
Timer expired send a new Path Delay Request      : 0
NeighborRateRatio has been computed successfully : 0
Path Delay has already been computed after init  : 1
Count consecutive reqs with multiple responses   : 0
Path Delay Response state machine variables:
Current state                                    : INITIAL_WAIT_REQ
SyncReceive state machine variables:
Current state                                    : DISCARD
A Sync Message has been received                 : no
A Follow Up Message has been received            : no
A Follow Up Message timeout                      : no
Time at which a Sync Message without Follow Up
                             will be discarded   : %lu
SyncSend state machine variables:
Current state                                    : INITIALIZING
A MDSyncSend structure has been received         : no
The timestamp for the sync msg has been received : no
PortSyncSyncReceive state machine variables:
Current state                                    : DISCARD
Grand Master / Local Clock frequency ratio       : 0.000000
A MDSyncReceive struct is ready to be processed  : no
Expiry of SyncReceiptTimeoutTimer                : no
PortSyncSyncSend state machine variables:
Current state                                    : TRANSMIT_INIT
Follow Up Correction Field of last recv PSS      : %ld
Upstream Tx Time of the last recv PortSyncSync   : %lu
Rate Ratio of the last received PortSyncSync     : 0.000000
GM Freq Change of the last received PortSyncSync : 0.000000
GM Time Base Indicator of last recv PortSyncSync : 0
Received Port Number of last recv PortSyncSync   : 0
PortSyncSync structure is ready to be processed  : no
Flag when the half_sync_itv_timer has expired    : no
Has half_sync_itv_timer expired twice            : no
Has syncReceiptTimeoutTime expired               : no
PortAnnounceReceive state machine variables:
Current state                                    : RECEIVE
An announce message is ready to be processed     : no
PortAnnounceInformation state machine variables:
Current state                                    : POST_DISABLED
Expired announce information                     : yes
PortAnnounceTransmit state machine variables:
Current state                                    : POST_IDLE
Trigger announce information                     : no

Statistics:
Sync messages received                 : 0
Follow Up messages received            : 0
Path Delay Request messages received   : 0
Path Delay Response messages received  : 0
Path Delay messages threshold exceeded : 0
Path Delay Follow Up messages received : 0
Announce messages received             : 0
ptp messages discarded                 : 0
Sync reception timeout                 : 0
Announce reception timeout             : 0
Path Delay Requests without a response : 173
Sync messages sent                     : 0
Follow Up messages sent                : 0
Path Delay Request messages sent       : 174
Path Delay Response messages sent      : 0
Path Delay Response FUP messages sent  : 0
Announce messages sent                 : 0
```

Reply:
@hakehuang could you please log an issue about this gPTP failure as it exists on the master branch as well.

@MaureenHelm Can we proceed with update to SDK 2.8. All networking tests are passing except for gptp

Reply:
@mmahadevan108  https://github.com/zephyrproject-rtos/zephyr/issues/29599 created

Reply:
@jukkar It appears that the gPTP failure also exists on master. Would you be ok approving this PR and handling the gPTP issue separately in #29599?

Reply:
@MaureenHelm the issue with gPTP is only seen in frdm-k64f board. gPTP works ok in Atmel sam-e70 (I tested it few days ago). I do not mind you merging this, after all this is NXP board, but I will not give +1 either. The gPTP has been broken either because of some HAL update or some other earlier commit to mcux driver.

Reply:
@jukkar I have gone back to release 2.3 and still see the console issue. Can you please let us know when frdm-k64f last worked for you?

Reply:
@mmahadevan108 sorry, cannot recall last time I saw it working. Note that there seems to be two issues that are perhaps related to each other:
* console does not show output, although I am seeing something right now
* network interface is down even if I have network cable connected
```
** Booting Zephyr OS build zephyr-v2.4.0-1085-g007677e67645  ***


[00:00:00.003,000] <wrn> net_if: iface 0x20000f60 is down
[00:00:00.007,000] <inf> net_config: Initializing network
[00:00:00.007,000] <inf> net_config: Waiting interface 1 (0x20000f60) to be up...
[00:00:01.005,000] <wrn> net_gptp: Reset Pdelay requests
[00:00:01.008,000] <wrn> net_if: iface 0x20000f60 is down
uart:~$ 
```

The console does not respond to anything after this output, perhaps the device is hang at this point.

What I did to compile gptp sample:
```
cd samples/net/gptp/
cmake -B build -DBOARD=frdm_k64f .
make -C build flash
```